core/tests/verifier_cache.rs

```diff
@@ -42,20 +42,20 @@ fn test_verifier_cache_rangeproofs() {
 	// Check our output is not verified according to the cache.
 	{
 		let mut cache = cache.write();
-		let unverified = cache.filter_rangeproof_unverified(&[out]);
+		let unverified = cache.filter_output_unverified(&[out]);
 		assert_eq!(unverified, vec![out]);
 	}
 
 	// Add our output to the cache.
 	{
 		let mut cache = cache.write();
-		cache.add_rangeproof_verified(vec![out]);
+		cache.add_output_verified(vec![out]);
 	}
 
 	// Check it shows as verified according to the cache.
 	{
 		let mut cache = cache.write();
-		let unverified = cache.filter_rangeproof_unverified(&[out]);
+		let unverified = cache.filter_output_unverified(&[out]);
```

Comment:
Please add a test to prove that the inflation issue is not reintroduced. That is, use the same rangeproof with a different output commitment and make sure that the cache does not consider it already verified.